### DISCUSSION TITLE
Erro no curso Flutter - Álcool e Gasolina

### DISCUSSION THREAD

**Discussion by Drioppi** [balta-io/forum#67] [Tira Dúvidas] [answered]
Conferi meu código, está como o seu e está dando erro, vou colar a msg de erro para você poder me ajudar.

Error: Cannot run with sound null safety, because the following dependencies
don't support null safety:

 - package:flutter_masked_text

For solutions, see https://dart.dev/go/unsound-null-safety

FAILURE: Build failed with an exception.

* Where:
Script 'C:\src\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 1159

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command 'C:\src\flutter\bin\flutter.bat'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 32s
Running Gradle task 'assembleDebug'...                             34,0s
Exception: Gradle task assembleDebug failed with exit code 1

Depois da primeira vez que inseri na main o   new MoneyMaskedTextController(),  o aplicativo não rodou mais, tentei corrigir mas não consegui e continuei a aula pra corrigir no final, mas mesmo conferindo o código o erro não corrigiu
![Screenshot_1](https://user-images.githubusercontent.com/116727435/198020815-ea254ba1-161b-402e-a02b-4f45ebd0ed55.png)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Drioppi, está usando as mesmas versões de pacotes que o curso? Isto importa bastante...

**Reply by Drioppi** [reply]
Como verifico se estou utilizando as mesmas versões de pacote?

**Reply by andrebaltieri** [reply]
Elas ficam no arquivo `pubspec.yaml`

**Reply by JeanMRocha** [reply]
Estou com o mesmo problema pois o pacote do flutter_masked_text.dart não tem a atualização para o null safe. Tem outros 3 erros que deram no meu projeto pois as libs sofreram mudanças desde o lançamento do curso, uma é que em muitos locais ele manda colocar o const e em algumas variáveis ele manda trocar o var por final.  Ainda tem um problema no input.widget.dart que pede pra colocar como imutável mas se colocar não aceita e manda retirar e fica neste loop. Balta Grande curso e estou aprendendo muito mas precisaria atualizar algumas aulas lá no canal.  O problema está no pacote da máscara.
![dart álcoolgasolinaapp](https://user-images.githubusercontent.com/111359346/198858560-e5914678-370c-43ea-beb3-e28c106c2b4e.png)

**Comment by JeanMRocha** [comment]
Poderia me ajudar a adaptar o https://pub.dev/packages/brasil_fields, que tem tudo que a gente precisa, mas está dando erro ainda no controller pois eu não sei usar.

**Reply by JeanMRocha** [reply]
Passei um tempo estudando e consegui, pelo menos em parte, resolver o problema atualizando para outra dependência:
adicionei esta nova dependência
dependencies:
  flutter_masked_text2: ^0.9.1

e troquei o import por este, parou meu roblema com o nulll safe sem ter de refatorar o código.

import 'package:flutter_masked_text2/flutter_masked_text2.dart';

Página da solução:
https://pub.dev/packages/flutter_masked_text2/install

**Reply by Drioppi** [reply]
Obrigada, vou tentar essa solução
